src/mysql-util/src/bin/mysql-gen.rs

```diff
@@ -0,0 +1,200 @@
+        vec!["INSERT IGNORE INTO customers (first_name, last_name, email) VALUES ('mattie', 'materialize', 'mm@materialize.com')",],
+        vec!["INSERT IGNORE INTO customers (first_name, last_name, email) VALUES ('nicky', 'materialize', 'nm@materialize.com')",],
+        vec!["INSERT IGNORE INTO customers (first_name, last_name, email) VALUES ('pat', 'materialize', 'pm@materialize.com')",],
+        vec!["INSERT IGNORE INTO customers (first_name, last_name, email) VALUES ('quinn', 'materialize', 'qm@materialize.com')",],
+        vec!["INSERT IGNORE INTO customers (first_name, last_name, email) VALUES ('riley', 'materialize', 'rm@materialize.com')",],
+        vec!["INSERT IGNORE INTO customers (first_name, last_name, email) VALUES ('taylor', 'materialize', 'tm@materialize.com')",],
+        vec![r#"INSERT INTO products (name, description, weight)
+                VALUES ('kitchen sink', 'something with everything', '2000.0')"#,
+         "INSERT INTO products_on_hand (product_id, quantity) VALUES (LAST_INSERT_ID(), rand()*25 + rand()*25)"],
+        vec![r#"INSERT INTO products (name, description, weight)
+                VALUES ('socket wrench', '12mm ratcheting', '.25')"#,
+         "INSERT INTO products_on_hand (product_id, quantity) VALUES (LAST_INSERT_ID(), rand()*50 + rand()*50)"],
+        vec![r#"INSERT INTO products (name, description, weight)
+                VALUES ('socket wrench', '13mm ratcheting', '.25')"#,
+         "INSERT INTO products_on_hand (product_id, quantity) VALUES (LAST_INSERT_ID(), rand()*50 + rand()*50)"],
+        vec![r#"INSERT INTO products (name, description, weight)
+                VALUES ('socket wrench', '14mm ratcheting', '.25')"#,
+         "INSERT INTO products_on_hand (product_id, quantity) VALUES (LAST_INSERT_ID(), rand()*50 + rand()*50)"],
+        vec!["DELETE FROM addresses ORDER BY rand() LIMIT 1",],
+        vec!["DELETE FROM customers ORDER BY rand() LIMIT 1",],
+        vec!["DELETE FROM orders ORDER BY rand() LIMIT 1",],
+        vec!["DELETE FROM products ORDER BY rand() LIMIT 1",],
+        vec!["DELETE FROM products_on_hand ORDER BY rand() LIMIT 1",],
+    ];
+
+    let handles = (0..8).map(|i| {
+        spawn({
+            let pool = pool.clone();
+            let queries = random_transactions.clone();
+
+            move || {
+                let mut conn = pool.get_conn()?;
+                let mut rng = thread_rng();
+
+                let mut successes = 0;
+                let mut failures = 0;
+
+                for _ in 0..10 {
+                    // Grab random set of queries and execute them. Ignore errors
+                    let mut tx = conn.start_transaction(TxOpts::default())?;
+
+                    let mut commit = true;
+
+                    // Randomly join multiple statements into the same transaction
+                    'transaction: for _ in 0..rng.gen_range(4..=8) {
```

Comment:
Might be worth extracting that second range into a named constant or something - I looked around whether the 8 had any special meaning (I don't think it does?)

Reply:
I can make these parameters instead of magic constants. I was simply playing around with these to change up the timing.

Reply:
I've gone ahead and made these parameters!